### PR TITLE
[DI] Remove deprecated dumping an uncompiled container

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | should be
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

See #20634

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
missing CHANGELOG updates in components (please check any your other PRs also if any)

**Comment by ro0NL** [comment]
@nicolas-grekas like so https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/CHANGELOG.md#300

can be done at once for all components right?

**Comment by nicolas-grekas** [comment]
or like so: https://github.com/symfony/symfony/pull/22779/files
it depends, we need to be as helpful to the reader as possible, case by case

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -154,17 +155,11 @@ public function dump(array $options = array())
...
-        } else {
-            $code .= $this->addConstructor();
-        }
-
-        $code .=
+        $code =
+            $this->startClass($options['class'], $options['base_class'], $options['namespace']).
+            $this->addFrozenConstructor().
+            $this->addFrozenCompile().
+            $this->addFrozenIsCompiled().
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: I suggest renaming these, remove "Frozen"
(+ rebase meanwhile :) )

**Comment by ro0NL** [comment]
Ready!

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Pull request merged**